File: jiaoyu-teacher-interface/src/main/java/com/jiaoyu/teacher/service/AsklistService.java
package com.jiaoyu.teacher.service;
import java.util.List;
import java.util.Map;

import com.jiaoyu.pojo.Asklist;

import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface AsklistService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Asklist> findAll();
	
	
	public Map<String,Object>  findAskListByKeId(Integer keid,Integer pageNum,Integer pageSize);
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Asklist asklist);
	
	
	/**
	 * 修改
	 */
	public void update(Asklist asklist);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Asklist findOne(Integer id);
	
	
	
	/**
	 * 根据用户名字找问题对象！！
	 * @param id
	 * @return
	 */
	public List<Map<String,Object>>  findAskByUsername(String username);
	
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Integer [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Asklist asklist, int pageNum,int pageSize);
	
}

File: jiaoyu-teacher-web/src/main/webapp/js/service/asklistService.js
//服务层
app.service('asklistService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../asklist/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../asklist/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../asklist/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../asklist/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../asklist/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../asklist/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../asklist/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});
